columnflow/plotting/plot_functions_2d.py

```diff
@@ -11,7 +11,6 @@
 import law
 
 from columnflow.util import maybe_import
-from columnflow.columnar_util import EMPTY_FLOAT
```

Comment:
One very stupid question just for my own education since I'm not an expert in the plotting. In many cases, the default value in case e.g. an observable cannot be calculated in a meaningful way is exactly the `EMPTY_FLOAT`. These values are then masked in the plotting. However, I don't see this here anywhere (also not in the `CreateHistogram` task). Since nothing in the previous version looks like it would do this as well, I assume that there are no issues with this new implementation. But where is the masking of e.g. the `EMPTY_FLOAT` values actually happening? Maybe @mafrahm can shed some light on this for me.

Reply:
@mafrahm can comment on the rationale of using `EMPTY_FLOAT` here, but to prevent any confusion I think it's important to note that missing values can arise in two very different contexts.

The first is when the value of a variable *in the underlying arrays* is missing or cannot be calculated in a meaningful way (e.g. leading muon p<sub>T</sub> for an event that has no muons). Here, `EMPTY_FLOAT = -99999.0` is stored by convention, since it lies outside the meaningful range of most variables. When filling a histogram, these entries end up in the under/overflow bins. 

The other case concerns empty bins in a histogram. This happens if there *are no events* where the variable being filled has a value falling within the range of the bin, and is *not* related to the occurrence of `EMPTY_FLOAT` in the input array. Indeed, there is no need to use `EMPTY_FLOAT` past the point where histograms are filled, but I'm guessing @mafrahm chose this as a marker for these bins because it's already used in representing missing values in the other case, which may have caused the confusion. Notwithstanding, I think `np.nan` is a better choice as an empty bin marker in histograms because `matplotlib` natively understands this as a missing value and masks the corresponding bins in the plot.

Reply:
I see, thank you very much for the clarification! It's all clear to me know. Personally, I absolutely agree that `np.nan` is a more natural choice, but we'll see what other people say.